### DISCUSSION TITLE
reached low water mark for data device - slow down the filesystem to unusability

### DISCUSSION THREAD

**Discussion by ingothierack** [stratis-storage/project#442] [Q&A]
maybe this already mentioned in another ticket but didn't found so feel free to close as duplicate.
If i stress a stratis filesystem with a lot of extensions, or say fill up with files:
`Mar 15 20:15:26 nas001 kernel: device-mapper: thin: 253:20: reached low water mark for data device: sending event.`
meaning in this case, copy a lot of files to the filesystem. All other filesystems in the pools are in case of this message(extending the thin pool) are very slow in response. Read operations(serving file over http or nfs or smb) are timed out(tooks longer than 10 seconds) and so one. Also the copy operation to this file system stops for a few seconds, when this message appears.
I'm here on normal spindle disks(SATA-devices)  in a software raid5(md device).
Maybe see some more information in Ticket stratis-storage/stratisd#1456 where i describe another problem.

Maybe there are some advice settings to avoid this.
Things currently set.
set scheduler to deadline for the raid disks
sysctl.conf settings:
vm.vfs_cache_pressure=50
vm.dirty_ratio = 6 
vm.dirty_background_ratio = 3
dev.raid.speed_limit_min=100000
dev.raid.speed_limit_max=500000

maybe there more or other optimize settings than that.
What i like to avoid, is that's the filesystem is complete unresponsive.

**Comment by mulkieran** [comment]
We've done many improvements in the thinpool layer and very likely this problem will have been resolved. Moving to project/Questions
